services/motion/explore/explore.go

```diff
@@ -0,0 +1,575 @@
+	return nil
+}
+
+type explore struct {
+	resource.Named
+
+	frameSystem referenceframe.FrameSystem
+	fsService   framesystem.Service
+	components  map[resource.Name]resource.Resource
+	services    map[resource.Name]resource.Resource
+	logger      golog.Logger
+	lock        sync.Mutex
+
+	obstacleResponseChan  chan moveResponse
+	executionResponseChan chan moveResponse
+	backgroundWorkers     *sync.WaitGroup
+}
+
+func (ms *explore) MoveOnMap(
+	ctx context.Context,
+	componentName resource.Name,
+	destination spatialmath.Pose,
+	slamName resource.Name,
+	extra map[string]interface{},
+) (bool, error) {
+	return false, errUnimplemented
+}
+
+func (ms *explore) MoveOnGlobe(
+	ctx context.Context,
+	componentName resource.Name,
+	destination *geo.Point,
+	heading float64,
+	movementSensorName resource.Name,
+	obstacles []*spatialmath.GeoObstacle,
+	motionCfg *motion.MotionConfiguration,
+	extra map[string]interface{},
+) (bool, error) {
+	return false, errUnimplemented
+}
+
+func (ms *explore) MoveOnGlobeNew(
+	ctx context.Context,
+	req motion.MoveOnGlobeReq,
+) (string, error) {
+	return "", errUnimplemented
+}
+
+func (ms *explore) GetPose(
+	ctx context.Context,
+	componentName resource.Name,
+	destinationFrame string,
+	supplementalTransforms []*referenceframe.LinkInFrame,
+	extra map[string]interface{},
+) (*referenceframe.PoseInFrame, error) {
+	return nil, errUnimplemented
+}
+
+func (ms *explore) StopPlan(
+	ctx context.Context,
+	req motion.StopPlanReq,
+) error {
+	return errUnimplemented
+}
+
+func (ms *explore) ListPlanStatuses(
+	ctx context.Context,
+	req motion.ListPlanStatusesReq,
+) ([]motion.PlanStatusWithID, error) {
+	return nil, errUnimplemented
+}
+
+func (ms *explore) PlanHistory(
+	ctx context.Context,
+	req motion.PlanHistoryReq,
+) ([]motion.PlanWithStatus, error) {
+	return nil, errUnimplemented
+}
+
+func (ms *explore) Close(ctx context.Context) error {
+	utils.FlushChan(ms.obstacleResponseChan)
+	utils.FlushChan(ms.executionResponseChan)
+	ms.backgroundWorkers.Wait()
+	return nil
+}
+
+// Move takes a goal location and will plan and execute a movement to move a component specified by its name
+// to that destination.
+func (ms *explore) Move(
+	ctx context.Context,
+	componentName resource.Name,
+	destination *referenceframe.PoseInFrame,
+	worldState *referenceframe.WorldState,
+	constraints *servicepb.Constraints,
+	extra map[string]interface{},
+) (bool, error) {
+	operation.CancelOtherWithLabel(ctx, exploreOpLabel)
+
+	// Parse extras
+	motionCfg, err := parseMotionConfig(extra)
+	if err != nil {
+		return false, err
+	}
+
+	// obstacleDetectors
+	obstacleDetectors, err := ms.createObstacleDetectors(motionCfg)
+	if err != nil {
+		return false, err
+	}
+
+	// Create kinematic base
+	kb, err := ms.createKinematicBase(ctx, componentName, motionCfg)
+	if err != nil {
+		return false, err
+	}
+
+	// Create motionplan plan
+	planInputs, err := ms.createMotionPlan(ctx, kb, destination.Pose(), worldState, extra)
```

Comment:
[nit]: this is not safe to do.
Since our destination is a pose in frame what if its frame is not world? In that case, we would need to transform it accordingly. Here is what we do in builtin
(https://github.com/viamrobotics/rdk/blob/main/services/motion/builtin/builtin.go#L174)

Reply:
Or maybe the frame of the base?

Reply:
Changed createMotionPlan to take in a PoseInFrame and transform it into the world frame via:

```
	goalPose, err := ms.fsService.TransformPose(ctx, destination, fs.World().Name(), nil)
	if err != nil {
		return nil, err
	}
```